### PR TITLE
[HDInsight] Adding Swagger-generated SDK and tests

### PR DESCRIPTION
**Pull request by wawon-msft**
Reopening pull request 3790 with requested changes.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@azuresdkci retest this please

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Please use generate.cmd to generate your SDK. This will create a new meta data file that links your spec commit id with the generated sdk. This helps in long term to know what SDK was generated using which spec commit id

**Comment by wawon-msft** [comment]
Looks like there were some significant changes to the generated files. I'll need to rerun tests.

File: src/SDKs/HDInsight/Management.HDInsight/Management.HDInsight.csproj [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.HDInsight</PackageId>
+    <Description>Azure HDInsight Management SDK Library</Description>
+    <AssemblyName>Microsoft.Azure.Management.HDInsight</AssemblyName>
+    <VersionPrefix>3.1.0-preview</VersionPrefix>
```
Comment (shahabhijeet) [inline src/SDKs/HDInsight/Management.HDInsight/Management.HDInsight.csproj]: @wawon-msft the last nuget that is released on nuget.org is 2.0.7, why jump to 3.1.0? Why not 3.0.0?
Any particular reason to make it 3.1.0?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Everything looks good.
Please clarify the version that is being used for this release and also use generate.cmd to generate your sdk, this will create a meta data file .txt that is essential for the SDK.

File: src/SDKs/HDInsight/Management.HDInsight/Management.HDInsight.csproj (hunk 1 above)
Comment (wawon-msft) [reply src/SDKs/HDInsight/Management.HDInsight/Management.HDInsight.csproj]: That was a mistake on my part. I bumped it up from 3.0.0, but did not realize 3.0.0 wasn't released yet.

**Comment by wawon-msft** [comment]
The metadata was now added with the last commit, sorry about that.

**Review by shahabhijeet — APPROVED** [review]

File: src/SDKs/HDInsight/Management.HDInsight/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+[assembly: NeutralResourcesLanguage("en")]
+
+// Setting ComVisible to false makes the types in this assembly not visible
+// to COM components.  If you need to access a type in this assembly from
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
+[assembly: AssemblyTitle("Microsoft Azure HDInsight Management Library")]
+[assembly: AssemblyDescription("Provides management functionality for Microsoft Azure HDInsight.")]
+[assembly: AssemblyVersion("3.1.0.0")]
```
Comment (shahabhijeet) [inline src/SDKs/HDInsight/Management.HDInsight/Properties/AssemblyInfo.cs]: @wawon-msft you don't have to bump up the AssemblyVersion.
Only bumping AssemblyFileVersion to 3.0.0.0 will be enough.
I just now realized you have previous versions that you use to publish using Hyak.
So all you need is AssemblyFileVersion and AssemblyVersion to 3.0.0.0

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**